src/braket/emulators/emulator_passes/criteria/connectivity_criterion.py

```diff
@@ -0,0 +1,143 @@
+        directed: bool = True,
+    ):
+        """
+        args:
+            connectivity_graph (Union[Dict[int, List[int]], DiGraph]): Either a sparse matrix or
+            DiGraph representation of the device connectivity. Can be None if fully_connected is
+            true.
+
+            fully_connected (bool): If true, the all qubits in the device are connected.
+
+            num_qubits (int): The number of qubits in the device; if fully_connected is True,
+            this is used to create a complete graph with num_qubits nodes; ignored if
+            connectivity_graph is provided and fully_connected if False.
+
+            qubit_labels (Iterable[int]): A set of qubit labels; if fully_connected is True,
+            the qubits_labels are used as nodes of a fully connected topology; ignored if
+            connectivity_graph is provided and fully_connected if False.
+
+            directed (bool): Denotes if the connectivity graph is directed or undirected. If
+                the connectivity graph is undirected, this constructor attempts to fill in any
+                missing back edges.
+        """
+
+        if not (connectivity_graph or fully_connected):
+            raise ValueError(
+                "Either the connectivity_graph must be provided or fully_connected must be True."
+            )
+
+        if fully_connected:
+            if not ((num_qubits is None) ^ (qubit_labels is None)):
+                raise ValueError(
+                    "Either num_qubits or qubit_labels (NOT both) must be \
+                        provided if fully_connected is True."
+                )
+            self._connectivity_graph = complete_graph(
+                num_qubits if num_qubits else qubit_labels, create_using=DiGraph()
+            )
+        elif not isinstance(connectivity_graph, DiGraph):
+            try:
+                self._connectivity_graph = from_dict_of_lists(
+                    connectivity_graph, create_using=DiGraph()
+                )
+            except Exception as e:
+                raise ValueError(
+                    f"connectivity_graph must be a valid DiGraph or a dictionary\
+                        mapping integers (nodes) to a list of integers (adjancency lists): {e}"
+                )
+        else:
+            self._connectivity_graph = connectivity_graph
+
+        if not directed:
+            for edge in self._connectivity_graph.edges:
+                self._connectivity_graph.add_edge(edge[1], edge[0])
+
+    def validate(self, circuit: Circuit) -> None:
+        """
+        Verifies that any verbatim box in a circuit is runnable with respect to the
+        device connectivity definied by this criterion. If any sub-circuit of the
+        input circuit is verbatim, we validate the connectivity of all gate operations
+        in the circuit.
+
+        Args:
+            circuit (Circuit): The Braket circuit whose gate operations to
+                validate.
+        """
```

Comment:
Should also add a `Raises` section

Reply:
Added a Raises section to all methods where an exception is raised!